real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/SensorPlayer.java

```diff
@@ -96,108 +111,124 @@ public void run() {
-                        logicNotificationPub.publish(transmitMessage);
-                        break;
-                    case StateMessage.VERSION_ID:
-                        transmitMessage = translator.fromJson(sensorDataJson, StateMessage.class);
-                        break;
-                    case SteeringMeasurement.VERSION_ID:
-                        transmitMessage = translator.fromJson(sensorDataJson, SteeringMeasurement.class);
-                        steeringPub.publish(transmitMessage);
-                        break;
-                    case WheelAngleCommandMeasurement.VERSION_ID:
-                        transmitMessage = translator.fromJson(sensorDataJson, WheelAngleCommandMeasurement.class);
-                        break;
-                    case TERMINATING_VERSION_ID:
-                        new RobobuggyLogicNotification("Stopping playback, hit a STOP", RobobuggyMessageLevel.NOTE);
-                        break;
-                    default:
-                        transmitMessage = new BaseMessage() {
-                            @Override
-                            public String toLogString() {
-                                return null;
-                            }
-
-                            @Override
-                            public Message fromLogString(String str) {
-                                return null;
-                            }
-                        };
-                        new RobobuggyLogicNotification("Couldn't parse a sensor found in the file: " + versionID, RobobuggyMessageLevel.WARNING);
-                        break;
-                }
-
-                BaseMessage timeMessage = (BaseMessage) transmitMessage;
+                    BaseMessage timeMessage = (BaseMessage) transmitMessage;
 
-                if (currentSensorTimeInMillis != -1) {
-                    if (timeMessage != null) {
-                        currentSensorTimeInMillis = timeMessage.getTimestamp().getTime();
-                        long sensorTimeFromPrev = currentSensorTimeInMillis - sensorStartTimeInMilis;
+                    if (timeMessage == null) {
+                        continue;
+                    }
 
-                        if(sensorTimeFromPrev > 0){
-                            Thread.sleep(sensorTimeFromPrev);
+                    if (prevSensorTime == -1) {
+                        prevSensorTime = timeMessage.getTimestamp().getTime();
+                    }
+                    else {
+                        long currentSensorTime = timeMessage.getTimestamp().getTime();
+                        long timeDiff = currentSensorTime - prevSensorTime;
+                        if (timeDiff > 10) {
+                            Thread.sleep(timeDiff);
                         }
+                        prevSensorTime = currentSensorTime;
+                    }
+
+
+                    switch (versionID) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Switch statements should have a default label](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2039563512/issues/source?bid=2938665&fileBranchId=3128705#l208)